### PR TITLE
Site redirects

### PR DESCRIPTION
**Pull request by sbesson**
Addresses https://trello.com/c/w31gbG72/15-broken-redirects.

The website redirects are modified as follows:

- all `/site/support/*` redirects are now handled via redirect map rather than proxying using named capture
- new mapping are added for the various `/site/xxx` categories
- if no match to the new website is found, the map fallbacks to the corresponding page under `www-legacy`
- additional mapping are added for the `omero-blog` and `info` redirects 
- the redirect map applies to the following locations `/site*`, `/omero-blog*` and `/info*`
- the remaining proxy backends are switched to use HTTPS when applicable and `/static` is proxied from www-legacy to fix the CSS of the QA application
- the custom 404 page is configured using `openmicroscopy.nginx-proxy` 1.4.0 (see https://github.com/openmicroscopy/ansible-role-nginx-proxy/pull/6)

The unnecessary `hsts` Nginx configuration is also removed and the tests are moved into the `www` directory.

This PR should be tested from a deployment perspective and functionally by checking the redirect issues reported during the last round of testing should now be working.

### CODE REVIEW

**Comment by sbesson** [comment]
@pwalczysko @kennethgillen fd43f00 should include the remaining configuration to fix the schema documentation.

**Comment by pwalczysko** [comment]
@sbesson : is this the PR where you encouraged me to put the results of testing ?

**Comment by pwalczysko** [comment]
https://docs.google.com/spreadsheets/d/1rqZdGtJHsrpBp8tvH9YrqquCJJu0p7yb_-1aBs05rV0/edit#gid=57813226 (yellow box about flash player) might need some checking possibly. 

https://docs.google.com/spreadsheets/d/1rqZdGtJHsrpBp8tvH9YrqquCJJu0p7yb_-1aBs05rV0/edit#gid=1728867908 are the red blockers (three last urls to recheck)

Does this one https://docs.google.com/spreadsheets/d/1rqZdGtJHsrpBp8tvH9YrqquCJJu0p7yb_-1aBs05rV0/edit#gid=1854829548 (mailing lists precise redirect, not just community, see yellow box) be worth some effort ?

This one https://docs.google.com/spreadsheets/d/1rqZdGtJHsrpBp8tvH9YrqquCJJu0p7yb_-1aBs05rV0/edit#gid=1212966858 (loss of the about/who... with the north atlantic map might be more serious actually - one detected, but not hightlighted red.

**Comment by pwalczysko** [comment]
other than https://github.com/openmicroscopy/infrastructure/pull/292#issuecomment-315842053 I do not see any lingering problems with the links

**Comment by pwalczysko** [comment]
@sbesson : When will be your last changes deployed ?

**Comment by sbesson** [comment]
Thanks @pwalczysko, 
- the QA issue is most likely completely independent of this integration and something to fix at the application level
- the mailing list link might require some anchors. However we dropped the intermediate page containing only the 2 links so this might require thought. Not a blocker to deployment
- the three failing URLs (schema documentation) should be fixed with fd43f00 - to be retested tomorrow
- the `site/about/who-ome` is currently manually redirected to `about`. I will push an extra commit to redirect this link to www-legacy as well.

**Comment by sbesson** [comment]
@pwalczysko the latest state of this PR is now deployed on infra-testpr.

File: ansible/server-state-playbooks/www/www.yml [hunk 1]
```diff
@@ -121,66 +121,49 @@
...
 
     nginx_proxy_redirect_map:
     # by default redirect to the 404 page
     - match: default
       dest: /404.html
-
-    # top-level
+    - match: "~/info/?$"
+      dest: /
+    - match: "~/info/(?<link>.*)$"
```
Comment (kennethgillen) [inline ansible/server-state-playbooks/www/www.yml]: What's this syntax?

Comment (sbesson) [inline ansible/server-state-playbooks/www/www.yml]: named capture - see http://nginx.org/en/docs/http/ngx_http_map_module.html#map

**Comment by pwalczysko** [comment]
@sbesson commented on the Flash Player on the QA2 tab problem see https://trello.com/c/hdyzXSu1/755-non-flash-qa-upload#comment-596dd7a91da2f12354458c3a

**Comment by pwalczysko** [comment]
For the Schemas red blockers, retest passed, see [column H](https://docs.google.com/spreadsheets/d/1rqZdGtJHsrpBp8tvH9YrqquCJJu0p7yb_-1aBs05rV0/edit#gid=1728867908)

**Comment by pwalczysko** [comment]
Accepting the mailing lists explanation @sbesson .

Ready to merge FMPOV

**Comment by kennethgillen** [comment]
Thanks for all the testing work, @pwalczysko!

**Pull request merged**